core/modules/layout/layout.api.php

```diff
@@ -196,6 +196,191 @@ function hook_layout_renderer_info() {
 }
 
 /**
+ * Respond to a block being removed from a layout.
+ *
+ * This hook is invoked from Layout::removeBlock() after a block has been 
+ * removed from a layout.
+ * In the context of the Layout Builder User Interface, this hook is 
+ * executed when a user clicks the "Remove block" link in the dropbutton
+ * which displays the block edit options.
+ * In a custom module, if custom settings have been saved as a result of
+ * a hook_block_save() operation, hook_layout_block_remove() allows
+ * these settings to be deleted if the block is removed before the layout
+ * is saved. 
+ *
+ * @param Layout $layout
+ *   The layout object from which the block is being removed.
+ * @param Block $block
+ *   The block object which is being removed.
+ *
+ * @see hook_block_save()
+ * @see hook_block_add()
+ *
+ * @ingroup layout_api_hooks
+ */
+function hook_layout_block_remove(Block $block, Layout $layout) {
+  if ($block->delta == 'my_block') {
+    config_set('my_module', 'custom_settings', array());
+  }
+}
+
+/**
+ * Respond to a block being programmatically added to a layout.
+ *
+ * This hook is invoked from Layout::addBlock() after a block has been 
+ * programmatically added to a region in a layout.
+ * In the context of the Layout Builder User Interface, this hook does NOT run
+ * after a user has clicked the "Add Block" button to insert the block into a 
+ * region.
+ * To respond to a block being added via Layout Builder, use hook_block_save().
+ *
+ * @param Layout $layout
+ *   The layout object to which the block is being added.
+ * @param Block $block
+ *   The block object which is being added.
+ *
+ * @see hook_block_save()
+ * @see hook_block_remove()
+ *
+ * @ingroup layout_api_hooks
+ */
+function hook_layout_block_add(Block $block, Layout $layout) {
+  if ($block->delta == 'my_block') {
+    config_set('my_module', 'custom_settings', $layout->content[$block->uuid]);
+  }
+}
+
+/**
+ * Respond to a layout being reverted.
+ *
+ * Layouts can be reverted only if the configuration is provided
+ * by a module. Layouts created in the Layout Builder User Interface
+ * cannot be reverted. 
+ * A layout revert operation results in deletion of the existing
+ * layout configuration and replacement with the default configuration
+ * from the providing module. 
+ * This hook is invoked from Layout::Revert() after a layout has been 
+ * reverted and the new configuration has been inserted into the live
+ * config directory and the layout cache has been cleared.
+ *
+ * @param Layout $layout
+ *   The old layout object that has just been deleted.
+ *
+ * @ingroup layout_api_hooks
+ */
+function hook_layout_revert(Layout $old_layout) {
+  if ($old_layout == 'my_layout') {
```

Comment:
Should this be `$old_layout->name`?

Is there a reason why `$new_layout` isn't passed in here as well?